pandas/tests/arrays/categorical/test_api.py

```diff
@@ -337,11 +337,23 @@ def test_remove_categories(self):
         # inplace == True
         res = cat.remove_categories("c", inplace=True)
         tm.assert_categorical_equal(cat, new)
+
         assert res is None
 
-        # removal is not in categories
-        with pytest.raises(ValueError):
-            cat.remove_categories(["c"])
+    @pytest.mark.parametrize(
+        "removals", [
+            (["c"]),
+            (["c", np.nan]),
+            ("c"),
+            (["c", "c"])
+        ],
+    )
+    def test_remove_categories_raises(self, removals):
+        cat = Categorical(["a", "b", "a"])
+        message = "removals must all be in old categories: {'c'}"
```

Comment:
for safety can you add the `re.escape` back here.

Reply:
I've added that. Thanks.